src/index.js

```diff
@@ -63,7 +63,7 @@ export class Helmet extends Component {
   static displayName = 'Helmet';
 
   shouldComponentUpdate(nextProps) {
-    return !fastCompare(this.props, nextProps);
+    return !fastCompare(without(this.props, 'helmetData'), without(nextProps, 'helmetData'));
```

Comment:
The `helmetData` is a different object in memory across multiple `<Helmet>` components. The underlying `context` inside it is the same, but the wrapper is different.